ome/omero-iviewer
# Roi query support

Support ```/iviewer/?rois=1,2``` to open ROIs in iviewer.
Shows a thumbnail view of multiple ROIs.

To test:
 - Open iviewer and select some saved Shapes
 - Check the Link icon in top-right of the ROIs tab - (Button should only be enabled if 1 shape is selected)
 - Click the button. Link to `?shape=ID` should be shown and selected (if unsaved Shapes are selected, should see a suitable message).
 - Copy the link and paste into browser...
 - Page should open iviewer to show the shape
    - Test with shape on non-default Z/T and not on first 'page' (>500 ROIs) e.g. https://merge-ci.openmicroscopy.org/web/iviewer?shape=223406
 - Thumbnails on left should load as expected for single Image:
    - If shape is on Orphaned Image, or multiple parents - No thumbnails
    - If shape is on Dataset Image, show Dataset thumbnails
    - If shape is on SPW image, show Well thumbnails
 - Browser link/history should behave as for single Image:
    - Clicking the thumnail will update the URL to point to that image
 - Also test linking by ROI ID: `/iviewer/?roi=1`. The *first* shape will be selected.
 - Probably a good idea to test other URLs still work as expected (thumbnails & URL updates):
    - Test `/iviewer/?dataset=1`
    - Test `/iviewer/?images=1` and `?images=1,2`
    - Test default viewer URL `/webclient/img_detail/2/?dataset=2`

Fixes #308 

This requires a new release of `omero-web` with https://github.com/ome/omero-web/pull/157 and https://github.com/ome/omero-web/pull/159 and https://github.com/ome/omero-web/pull/178
 y
